### PR TITLE
Miscellaneous fixes and cleanup.

### PR DESCRIPTION
**Pull request by DanielJoyce**
In using three.dart for my 3d printing chrome app, I've had to implement various fixes and fix broken code. These are all simple bug fixes.

### CODE REVIEW

File: lib/src/scenes/fog_exp2.dart [hunk 1]
```diff
@@ -1,7 +1,8 @@
 part of three;
 
-class FogExp2 implements Fog {
-  Color color;
+class FogExp2 extends Fog {
   num density;
-  FogExp2(num hex, [this.density = 0.00025]) : color = new Color(hex);
+  FogExp2(num hex, [this.density = 0.00025]) {
+    color = new Color(hex);
```
Comment (beatgammit) [inline lib/src/scenes/fog_exp2.dart]: This line can be reverted. It looks like the main change was removing the local `Color color`.

File: lib/src/scenes/fog_linear.dart [hunk 2]
```diff
@@ -8,9 +8,10 @@ part of three;
...
  */
 
-class FogLinear implements Fog {
-  Color color;
+class FogLinear extends Fog {
   double near;
   double far;
-  FogLinear(num hex, [this.near = 1.0, this.far = 1000.0]) : color = new Color(hex);
+  FogLinear(num hex, [this.near = 1.0, this.far = 1000.0]) {
+    color = new Color(hex);
```
Comment (beatgammit) [inline lib/src/scenes/fog_linear.dart]: This line can be reverted. It looks like the main change was removing the local `Color color`.

File: lib/src/renderers/web_gl_renderer.dart [hunk 3]
```diff
@@ -6145,7 +6147,7 @@ class WebGLRenderer implements Renderer {
         spotLength += 1;
 
       } else if (light is HemisphereLight) {
-
+
```
Comment (beatgammit) [inline lib/src/renderers/web_gl_renderer.dart]: This should be removed. In fact, could you run `dartfmt --max_line_length 120` on the files you changed? I see a couple other lines that should probably be formatted.

File: lib/src/loaders/stl_loader.dart [hunk 4]
```diff
@@ -35,13 +35,22 @@ class STLLoader extends Loader {
   Future<Geometry> load(url) =>
       HttpRequest.request(url, responseType: "arraybuffer").then((req) => _parse(req.response));
 
-  _parse(Uint8List data) => _isBinary(data) ? _parseBinary(data.buffer) : _parseASCII(new String.fromCharCodes(data));
+  Geometry _parse(Object resp){
+    Uint8List data;
+    if(!(resp is Uint8List)){
```
Comment (beatgammit) [inline lib/src/loaders/stl_loader.dart]: This can be `if (resp is! Uint8List) {`.

**Comment by beatgammit** [comment]
Thanks for the fixes!

I left some comments on things that I think could be improved before merging. I also haven't actually tried running any of the changes, but everything looks pretty good to me.

Perhaps someone else should review the methods that you removed the `_` prefix from to make sure that those functions are good enough for a public API.

**Comment by DanielJoyce** [comment]
I exposed the methods in stl loader so I could create a subclass that loads
from a file handle. seems to work perfectly fine. Dart doesn't have any
notion of protected. Its either package private or public.

**Comment by johsin18** [comment]
The first commit is a merge commit, which is a bit strange.  Can you rebase on the latest master first, please?  I think some things had been fixed on master already.

**Comment by DanielJoyce** [comment]
Yeah, I'll look into doing that this weekend.

**Pull request closed without merging**